### ISSUE TITLE
Unreasonably slow view painting at HiDPI

### ISSUE THREAD

**Issue by EwanGreen4** [yuxshao/ptcollab#81]
Makes the app very slow on a high DPI screen (4k at fullscreen should be doable). Smooth playback requires the use of a very high buffer and framerates range from 5-35 depending on the project loaded (for me, at least; Win11 on i5 9400f and 1070 TI) . The painting APIs used are unable to effectively leverage the GPU for the things that would help in this case, since the slog is mainly logic-based and doesn't lie in the actual painting operations. Of all the QOL things being added to ptcollab recently this one seems like it should be a pretty obvious priority considering I find myself changing my desktop resolution just to use ptcollab & have it fill up my whole screen

**Comment by yuxshao** [comment]
This change is a bigger lift than the other things and adds quite a bit of complexity (or so I imagine) which is why it's been put off for a bit -- but I agree it's important. Is it any better if you zoom in a bunch when at hidpi?

**Comment by EwanGreen4** [comment]
>  (And it sounds like it's Ok at regular DPI, is that right?)
At 96 (standard) DPI I can only imagine it slowing down on really bad hardware. At which point pxtone would already be stuttering &/ high buffer is needed, probably

> Is it any better if you zoom in a bunch when at hidpi? 
The effect is lessened but still noticeable

**Comment by yuxshao** [comment]
okay some findings
1. drawing pixmaps is slow at higher dpi even if it's convenient (akin to drawing on photoshop layers and then merging them together), so should probably avoid that
2. the qbrush / qcolor conversion does seem to be the diff of a few fps
3. should probably prioritize a 2x view similar to ptweb to avoid rendering at unnecessarily high resolution

**Comment by yuxshao** [comment]
https://github.com/yuxshao/ptcollab/pull/83
I did 1 and 2 and I think that should get the bulk of it... I'll look again tomorrow before merging, but give it a shot if you want!

I looked into 3, but I think that there might not be that much savings. The thing I found was QPainter::setTransform which just changes the scale on the painter associated with the drawing targets (the piano roll widget). In the end that's the same as like, multiplying all the coords passed into drawRect by some constant, which is how the zooming works right now.
